source/slang/ir-specialize.cpp

```diff
@@ -1207,6 +1291,246 @@ struct SpecializationContext
         }
     }
 
+    UInt calcExistentialBoxSlotCount(IRType* type)
```

Comment:
Wouldn't a boring `while(true)` be 'better' - more normal/less gotoy.

Reply:
Yeah, it would probably be more canonically Good Programming Style to use an infinite loop instead of the `goto` here, but in some ways it would also obscure the intention. For somebody reading this code, it is best not to think in terms of a loop, but rather a simple function that works by case analysis (and just happens to have a tail-recursive case that we've optimized via `goto`).

I can make it a loop instead if we think that `goto`-avoidance is a worthwhile style rule to enforce (I know it can lead to warnings around skipping constructors when one combines `goto` and C++).